pkg/operator/ceph/file/filesystem.go

```diff
@@ -179,6 +178,10 @@ func (f *Filesystem) doFilesystemCreate(context *clusterd.Context, cephVersion c
 					fmt.Sprintf(": %+v", err),
 			)
 		}
+		if err := client.SetPools(context, cephVersion, clusterName, f.Name, f.dataPools, f.metadataPool); err != nil {
+
+			return nil
+		}
```

Comment:
Here I'll add condition if the update is not possible, I'll take references from this function https://github.com/rook/rook/blob/ab69d75cf2f5c6279b3f044493bf32e3c3a9c779/pkg/operator/ceph/file/filesystem.go#L174 to write mine.

Reply:
Ok.